src/bin/cargo-miri.rs

```diff
@@ -333,10 +333,11 @@ path = "lib.rs"
         None => true,
         Some(target) => target == rustc_version::version_meta().unwrap().host,
     };
-    let sysroot = if is_host { dir.join("HOST") } else { PathBuf::from(dir) };
+    let sysroot = if is_host { dir.join("HOST") } else { PathBuf::from(dir) }; 
+    
     std::env::set_var("MIRI_SYSROOT", &sysroot); // pass the env var to the processes we spawn, which will turn it into "--sysroot" flags
     if print_env {
-        println!("MIRI_SYSROOT={}", sysroot.display());
+        println!("MIRI_SYSROOT={:?}", &sysroot); // for Windows users, prints path with backslashes escaped.
```

Comment:
This is not just for Windows users. It just as much affects Linux users with `'`, `"` or `\` in the path (all of which are valid filename characters on Linux).

Reply:
You are right. Do you think I should add a macro for conditional compilation for Windows users??

Reply:
No, what I am saying is the bug you are fixing also affects other platforms. With conditional compilation you'd only fix it on Windows, why do you want that? :D